### ISSUE TITLE
Feature request: points to labels

### ISSUE THREAD

**Issue by jni** [scikit-image/scikit-image#5087]
## Description

I find myself repeating the following lines very often:

```python
coords = ... # something that gives me an (npoints, ndim) array
indices = tuple(np.transpose(np.round(coords).astype(int)))
labels = np.zeros_like(image, dtype=np.uint32)
labels[indices] = np.arange(1, len(coords) + 1)
```

This is required for example to take a set of centroids and turn them into markers for watershed.

I think it would be great for skimage to provide a utility function to do this.

**Comment by emmanuelle** [comment]
Good idea. Should it be in `segmentation` or in `utils`?

**Comment by jni** [comment]
A great question. I don't have a strong preference. Maybe we reopen #1608? =P

**Comment by harish-vnkt** [comment]
Hi, @jni. First timer in open source and would like to work on this issue. :) 
Would it be enough to have a function that takes centroid coordinates and the image as arguments and uses [ndi.label](https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.label.html#scipy.ndimage.label) to assign labels to the centroids? Also, any idea on which module this function belongs to?

**Comment by jni** [comment]
@harish-vnkt Hello! I think that approach will be slower than using the direct assignment in the above code snippet.

Regarding module, I'm not sure, nor do I know what the best name is. `skimage.util.points_to_labels`? `skimage.segmentation.points_to_label_image`? Some combination of the above? But, might as well open the PR and get the ball rolling! 😊

**Comment by jni** [comment]
Closed by #5180
